### PR TITLE
Add ForAllSuchThatDecider

### PR DESCRIPTION
**Pull request by syuhei176**
Adding ForAllSuchThatDecider using FreeVariable class. And clean tests using helper function.

### CODE REVIEW

**Comment by syuhei176** [comment]
split this PR

**Comment by syuhei176** [comment]
I splitted into https://github.com/cryptoeconomicslab/wakkanay/pull/47. please review it first.

File: src/ovm/deciders/operators/ForAllSuchThatDecider.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { DeciderManager } from '../../DeciderManager'
+
+/**
+ * ForDecider check quantifier and property.
+ */
+export class ForAllSuchThatDecider implements Decider {
+  public async decide(
+    manager: DeciderManager,
+    inputs: Bytes[],
+    substitutions: { [key: string]: Bytes } = {}
```
Comment (syuhei176) [inline src/ovm/deciders/operators/ForAllSuchThatDecider.ts]: @tkmct if we don't want to propagate substitutions, we should replace variables recursively.

Comment (tkmct) [reply src/ovm/deciders/operators/ForAllSuchThatDecider.ts]: I think it's okay to propagate substitutions for now.

**Pull request merged**